### PR TITLE
Add string table generator

### PR DESCRIPTION
**Pull request by paddatrapper**
Add string table generator to the library, instead of including it in each firmware repo individually

### CODE REVIEW

File: descriptors_string_table.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+
+import sys
+
+strings = [x.strip() for x in sys.stdin.readlines()]
+
+if sys.argv[1] == "--header":
```
Comment (mithro) [inline descriptors_string_table.py]: We should probably be using argparse here.

File: descriptors_string_table.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		__u8 bDescriptorType;
+		__le16 wData[1];
+	} language;""")
+	for i, string in enumerate(strings):
+		print("""\
+	struct usb_string_%(i)i {
+		__u8 bLength;
+		__u8 bDescriptorType;
+		__le16 wData[%(l)i];
+	} string%(i)i;""" % {'l': len(string), 'i': i})
```
Comment (mithro) [inline descriptors_string_table.py]: Should probably use `.format(` rather than `%`

File: descriptors_string_table.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+
+import sys
+
+strings = [x.strip() for x in sys.stdin.readlines()]
+
+if sys.argv[1] == "--header":
+	print("""\
+// This is an auto-generated file!
```
Comment (mithro) [inline descriptors_string_table.py]: This comment could probably use some more info about what script generated it, what the input was, etc.

File: descriptors_string_table.py (hunk 1 above)
Comment (paddatrapper) [reply descriptors_string_table.py]: Python is not my strong suite. I have used argparse in the past, but always struggled with it

Comment (mithro) [reply descriptors_string_table.py]: The docs for it seem pretty good?

**Comment by paddatrapper** [comment]
Fixed

**Review by mithro — COMMENTED** [review]
Can you make this file pass pep8?

**Comment by paddatrapper** [comment]
Fixed

File: descriptors_string_table.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python3
+
+import argparse
+import sys
+
+parser = argparse.ArgumentParser(
+    description="Generate a C file containing strings in the format needed " +
+    "by the USB descriptors")
+parser.add_argument("--header",
```
Comment (mithro) [inline descriptors_string_table.py]: I changed the wrapping of these.

**Comment by mithro** [comment]
Merged.

**Pull request merged**